posthog/tasks/alerts/trends.py

```diff
@@ -0,0 +1,216 @@
+from typing import Optional, cast
+
+from posthog.api.services.query import ExecutionMode
+from posthog.caching.calculate_results import calculate_for_query_based_insight
+
+from posthog.models import AlertConfiguration, Insight
+from posthog.schema import (
+    TrendsQuery,
+    IntervalType,
+    TrendsAlertConfig,
+    InsightThreshold,
+    AlertCondition,
+    AlertConditionType,
+    InsightsThresholdBounds,
+    InsightThresholdType,
+)
+from posthog.caching.fetch_from_cache import InsightResult
+from typing import TypedDict, NotRequired
+from posthog.tasks.alerts.utils import (
+    AlertEvaluationResult,
+    NON_TIME_SERIES_DISPLAY_TYPES,
+)
+
+
+# TODO: move the TrendResult UI type to schema.ts and use that instead
+class TrendResult(TypedDict):
+    action: dict
+    actions: list[dict]
+    count: int
+    data: list[float]
+    days: list[str]
+    dates: list[str]
+    label: str
+    labels: list[str]
+    breakdown_value: str | int | list[str]
+    aggregated_value: NotRequired[float]
+    status: str | None
+    compare_label: str | None
+    compare: bool
+    persons_urls: list[dict]
+    persons: dict
+    filter: dict
+
+
+def check_trends_alert(alert: AlertConfiguration, insight: Insight, query: TrendsQuery) -> AlertEvaluationResult:
+    if "type" in alert.config and alert.config["type"] == "TrendsAlertConfig":
+        config = TrendsAlertConfig.model_validate(alert.config)
+    else:
+        ValueError(f"Unsupported alert config type: {alert.config}")
+
+    condition = AlertCondition.model_validate(alert.condition)
+    threshold = InsightThreshold.model_validate(alert.threshold.configuration) if alert.threshold else None
+
+    if not threshold:
+        return AlertEvaluationResult(value=0, breaches=[])
+
+    match condition.type:
+        case AlertConditionType.ABSOLUTE_VALUE:
+            if threshold.type != InsightThresholdType.ABSOLUTE:
+                raise ValueError(f"Absolute threshold not configured for alert condition ABSOLUTE_VALUE")
+
+            # want value for current interval (last hour, last day, last week, last month)
+            # depending on the alert calculation interval
+            if _is_non_time_series_trend(query):
+                filters_override = _date_range_override_for_intervals(query)
```

Comment:
How does this interact with when the new interval rolls over? Does `_date_range_override_for_intervals` always look at the current interval?

For example, if the interval size is DAY and this alert is set to run at 1 AM, will it always be looking at an interval from midnight to 1AM or do we do something else here?

Reply:
Yes it always looks at the current interval for absolute errors.
Here, it will always look from `midnight to 1AM` and they would need to put it as an hourly check to get the day value. 

Otherwise we could enforce that we report absolute errors on a lag of one interval, like in this case check the value for the previous day (which won't change anymore), but then if it's an hourly alert check, it will keep alerting for yesterday (if yesterday was breaching)